numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2253,8 +2253,9 @@ array_vdot(PyObject *NPY_UNUSED(dummy), PyObject *args)
 {
     int typenum;
     char *ip1, *ip2, *op;
-    npy_intp n, stride;
+    npy_intp n, stride1, stride2;
     PyObject *op1, *op2;
+    PyArray_Dims newdims = {{-1}, 1};
```

Comment:
I randomly tried, and it compiled, so I thought it probably would.... Will upload the fixes in 5 minutes.